bin/hdfcoinc/pycbc_fit_sngls_split_binned

```diff
@@ -0,0 +1,354 @@
+    # Initialize statclass with an empty file list. In general could feed it
+    # files here for statistics which need that.
+    stat_instance = sngl_statistic_dict[statchoice]([])
+    return stat_instance.single(trigs)
+
+parser = argparse.ArgumentParser(usage="",
+    description="plot histograms of triggers split over various")
+parser.add_argument("--trigger-file", required=True,
+                    help="Input hdf5 file containing single triggers. "
+                    "Required")
+parser.add_argument("--bank-file", default=None, required=True,
+                    help="hdf file containing template parameters. Required")
+parser.add_argument('--output-file', type=str, required=True,
+                    help="Output image file. Required")
+parser.add_argument('--verbose', action='store_true')
+parser.add_argument('--bin-param', default='template_duration', type=str,
+                    help="parameter for on-plot splitting. Default: template_duration")
+parser.add_argument('--num-bins', type=int, default=6,
+                    help="number of bins over which to split bin-param. Default: 6")
+parser.add_argument('--max-bin-param',type=float, default=None,
+                    help="maximum allowed value of bin-param")
+parser.add_argument('--min-duration', type=float, default=None,
+                    help="Fudge factor for templates with tiny or negative "
+                        "values of template_duration: add to duration values "
+                        "before fitting. Units seconds")
+parser.add_argument('--split-param-one', default='eta', type=str,
+                    help="Parameter over which to split plot grid in the "
+                         "y direction. Default: eta")
+parser.add_argument('--split-param-two', default='chi_eff', type=str,
+                    help="Parameter over which to split plot grid in the "
+                         "x direction. Default: chi_eff")
+parser.add_argument('--split-one-nbins', default=3, type=int,
+                    help="number of bins over which to split split-param-one. "
+                         "Default: 3")
+parser.add_argument('--split-two-nbins', default=4, type=int,
+                    help="number of bins over which to split split-param-two. "
+                         "Default: 4")
+parser.add_argument('--split-one-log', action='store_true', default=False,
+                    help="is split-param-one split in log-space?. Default: False")
+parser.add_argument('--split-two-log', action='store_true', default=False,
+                    help="is split-param-two split in log-space?. Default: False")
+parser.add_argument("--sngl-stat", default="new_snr",
+                    choices=sngl_statistic_dict.keys(),
+                    help="Function of SNR and chisq to perform fits with")
+parser.add_argument('--ifo', type=str, help="Interferometer")
+parser.add_argument('--plot-max-x',default=1000,
+                    help="Maximum stat value for use in plots, if this "
```

Comment:
this seems overcomplicated /backwards in terms of how it is interpreted by the code. 
would it not work to 1) if no plot-max-x is given (ie None), scale plots to 1.05* largest stat 
2) if plot-max-x is given, use that value?

Reply:
Agree - I think this was the simpler option when I was performing the testing for some reason. sorted now